bindings/gumjs/gumv8memory.cpp

```
@@ -349,6 +351,29 @@ GUMJS_DEFINE_FUNCTION (gumjs_memory_protect)
   info.GetReturnValue ().Set (success);
 }
 
+GUMJS_DEFINE_FUNCTION (gumjs_memory_query_protection)
+{
+  gpointer address;
+  GumPageProtection prot;
+
+  if (!_gum_v8_args_parse (args, "p", &address))
+    return;
+
+  if (!gum_memory_query_protection (address, &prot))
+    return;
```

Need to throw a JS exception here.